examples/qml_features/rust/src/nested_qobjects.rs

```diff
@@ -0,0 +1,57 @@
+// SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+#[cxx_qt::bridge(cxx_file_stem = "nested_qobjects")]
+mod ffi {
+    unsafe extern "C++" {
+        #[cxx_name = "InnerObject"]
+        type InnerObjectPtr = super::qobject::InnerObject;
+    }
+
+    #[cxx_qt::qobject]
+    #[derive(Default)]
+    pub struct InnerObject {
+        #[qproperty]
+        counter: i32,
+    }
+
+    #[cxx_qt::qobject]
+    pub struct OuterObject {
+        #[qproperty]
+        inner: *mut InnerObjectPtr,
+    }
+
+    impl Default for OuterObject {
+        fn default() -> Self {
+            Self {
+                inner: std::ptr::null_mut(),
+            }
+        }
+    }
+
+    #[cxx_qt::qsignals(OuterObject)]
+    pub enum OuterSignals {
+        Called { inner: *mut InnerObjectPtr },
+    }
+
+    impl qobject::OuterObject {
+        // InnerObjectPtr needs to be called InnerObject in our C++ generation...
+        #[qinvokable]
+        pub unsafe fn invokable(self: Pin<&mut Self>, inner: *mut InnerObjectPtr) {
+            if let Some(inner) = inner.as_ref() {
+                println!("Inner object: {}", inner.counter());
+            }
+
+            self.emit(OuterSignals::Called { inner });
+        }
+
+        #[qinvokable]
+        pub fn reset(self: Pin<&mut Self>) {
+            if let Some(inner) = unsafe { self.inner().as_mut() } {
```

Comment:
To reach the getters and setters you need to get from a `*mut T` to a `Pin<&mut T>` which then requires some `unsafe` calls.

If the property was `Pin` already would this help ?

Reply:
Done